File: app/src/main/java/com/example/a59682/myapplicationtest2018421/PhotoAdapter.java
package com.example.a59682.myapplicationtest2018421;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import java.util.List;

/**
 * @author 59682
 * @name PhotoAdapter
 * @description adapter-recyclerview
 */
public class PhotoAdapter extends RecyclerView.Adapter<PhotoAdapter.ViewHolder> {

    private Context mContext;

    private List<Photo> mPhotoList;

    /**
     * @author 59682
     * @name ViewHolder
     * @description 嵌套缓存类，直接从Holder中取item 无需findid重新加载
     */
    static class ViewHolder extends RecyclerView.ViewHolder {

        /* view点击事件 */
        CardView cardView;
        ImageView photoImage;
        TextView photoName;

        public ViewHolder(View itemView) {
            super(itemView);
            cardView = (CardView) itemView;
            photoImage = itemView.findViewById(R.id.photo_image);
            photoName = itemView.findViewById(R.id.photo_name);
        }
    }

    public PhotoAdapter(List<Photo> photoList) {
        mPhotoList = photoList;
    }

    /**
     * @description 生成Holder实例
     */
    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        if (mContext == null) {
            mContext = parent.getContext();
        }
        View view = LayoutInflater.from(mContext).inflate(R.layout.photo_item, parent, false);
        return new ViewHolder(view);
    }

    /**
     * @description 滚动到屏幕内时执行
     */
    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Photo photo = mPhotoList.get(position);

        holder.photoName.setText(photo.getName());

        /* 使用Glide加载图片 */
        Glide.with(mContext).load(photo.getImageId()).into(holder.photoImage);
    }

    /**
     * @description 返回item总的项数
     */
    @Override
    public int getItemCount() {
        return mPhotoList.size();
    }
}

File: app/src/main/java/com/ten/lifecat/phone/view/adapter/PhotoAdapter.kt
package com.ten.lifecat.phone.view.adapter

import android.content.Context
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide
import com.ten.lifecat.phone.model.bean.Photo
import com.ten.lifecat.phone.R

/**
 * 图片卡片适配器 adapter for recyclerView
 */
class PhotoAdapter(private val mPhotoList: List<Photo>) : RecyclerView.Adapter<PhotoAdapter.ViewHolder>() {

    private var mContext: Context? = null

    /**
     * 嵌套缓存类，直接从Holder中取item 无需findid重新加载
     */
    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        /* view点击事件 */
        var cardView: CardView = itemView as CardView
        var photoImage: ImageView = itemView.findViewById(R.id.photo_image)
        var photoName: TextView = itemView.findViewById(R.id.photo_name)
    }

    /**
     * 生成Holder实例
     */
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        if (mContext == null) {
            mContext = parent.context
        }
        val view = LayoutInflater.from(mContext).inflate(R.layout.photo_item, parent, false)
        return ViewHolder(view)
    }

    /**
     * 滚动到屏幕内时执行
     */
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val photo = mPhotoList[position]

        holder.photoName.text = photo.name

        /* 使用Glide加载图片 */
        Glide.with(mContext!!).load(photo.imageId).into(holder.photoImage)
    }

    /**
     * 返回item总的项数
     */
    override fun getItemCount(): Int {
        return mPhotoList.size
    }
}
